### PR TITLE
SB-8241 - changes for curernt branch.

### PR DESCRIPTION
**Pull request by amolnin**
discarding other changes from unit-test done earlier for test-case
re-factoring.

### CODE REVIEW

File: common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java [hunk 1]
```diff
@@ -24,7 +24,7 @@ private UserRequestValidator() {}
    *
    * @param userRequest Request
    */
-  public static void validateCreateUser(Request userRequest) {
+  public static void validateCreateUserRequest(Request userRequest) {
```
Comment (bvinayakumar) [inline common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java]: Remove static from all these methods.

File: common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java [hunk 2]
```diff
@@ -357,7 +361,7 @@ private static boolean validatePhoneNo(String phone, String countryCode) {
    * @param userRequest Request
    */
   @SuppressWarnings({"rawtypes"})
-  public static void validateUpdateUser(Request userRequest) {
+  public static void validateUpdateUserRequest(Request userRequest) {
```
[the remark this answers is not in the document]
Comment (bvinayakumar) [reply common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java]: Resolve above codacy error.

File: common-util/src/main/java/org/sunbird/common/request/UserProfileRequestValidator.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+  private void checkExistenceAndDataTypeForPublicPrivate(Request request) {
+    if (request.getRequest().get(JsonKey.PRIVATE) == null
+        && request.getRequest().get(JsonKey.PUBLIC) == null) {
+      throw new ProjectCommonException(
+          ResponseCode.invalidData.getErrorCode(),
+          ResponseCode.invalidData.getErrorMessage(),
+          ResponseCode.CLIENT_ERROR.getResponseCode());
+    }
+    checkNotInstanceOfList(request, JsonKey.PRIVATE);
```
Comment (bvinayakumar) [inline common-util/src/main/java/org/sunbird/common/request/UserProfileRequestValidator.java]: Use BaseRequestValidator.validateListParam.

File: common-util/src/main/java/org/sunbird/common/request/UserProfileRequestValidator.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    checkNotInstanceOfList(request, JsonKey.PUBLIC);
+  }
+
+  private void checkNotInstanceOfList(Request request, String forKey) {
```
Comment (bvinayakumar) [inline common-util/src/main/java/org/sunbird/common/request/UserProfileRequestValidator.java]: Not needed.

File: common-util/src/main/java/org/sunbird/common/request/UserProfileRequestValidator.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+import org.sunbird.common.exception.ProjectCommonException;
+import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.models.util.ProjectUtil;
+import org.sunbird.common.responsecode.ResponseCode;
+
+public class UserProfileRequestValidator extends BaseRequestValidator {
+
+  @SuppressWarnings("unchecked")
+  public void validateProfileVisibility(Request request) {
+    if (request.getRequest().get(JsonKey.USER_ID) == null
```
Comment (bvinayakumar) [inline common-util/src/main/java/org/sunbird/common/request/UserProfileRequestValidator.java]: BaseRequestValidator.validateParam can be used with mandatoryParamsMissing error as there is no check being done on username.

File: common-util/src/main/java/org/sunbird/common/request/UserProfileRequestValidator.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        || StringUtils.isBlank(((String) request.getRequest().get(JsonKey.USER_ID)))) {
+      throw new ProjectCommonException(
+          ResponseCode.usernameOrUserIdError.getErrorCode(),
+          ResponseCode.usernameOrUserIdError.getErrorMessage(),
+          ResponseCode.CLIENT_ERROR.getResponseCode());
+    }
+    validateUserId(request, JsonKey.USER_ID);
+    checkExistenceAndDataTypeForPublicPrivate(request);
```
Comment (bvinayakumar) [inline common-util/src/main/java/org/sunbird/common/request/UserProfileRequestValidator.java]: checkExistenceAndDataTypeForPublicPrivate -> validatePublicAndPrivateFields

File: common-util/src/main/java/org/sunbird/common/request/UserProfileRequestValidator.java [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+          ResponseCode.CLIENT_ERROR.getResponseCode());
+    }
+  }
+
+  private void checkKeyNotPresentInPublicPrivate(Request request) {
+    if (null != request.getRequest().get(JsonKey.PRIVATE)
+        && null != request.getRequest().get(JsonKey.PUBLIC)) {
+      List<String> privateList = (List<String>) request.getRequest().get(JsonKey.PRIVATE);
+      List<String> publicList = (List<String>) request.getRequest().get(JsonKey.PUBLIC);
+      if (privateList.size() > publicList.size()) {
```
Comment (bvinayakumar) [inline common-util/src/main/java/org/sunbird/common/request/UserProfileRequestValidator.java]: If condition is not required.

File: common-util/src/main/java/org/sunbird/common/request/UserProfileRequestValidator.java [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+        checkNotExistsInBothList(publicList, privateList);
+      } else {
+        checkNotExistsInBothList(privateList, publicList);
+      }
+    }
+  }
+
+  private void checkNotExistsInBothList(List<String> checkInList, List<String> iterateList) {
```
Comment (bvinayakumar) [inline common-util/src/main/java/org/sunbird/common/request/UserProfileRequestValidator.java]: checkInList -> list1
iterateList -> list2

File: common-util/src/main/java/org/sunbird/common/request/UserProfileRequestValidator.java [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+    }
+  }
+
+  private void checkKeyNotPresentInPublicPrivate(Request request) {
+    if (null != request.getRequest().get(JsonKey.PRIVATE)
+        && null != request.getRequest().get(JsonKey.PUBLIC)) {
+      List<String> privateList = (List<String>) request.getRequest().get(JsonKey.PRIVATE);
+      List<String> publicList = (List<String>) request.getRequest().get(JsonKey.PUBLIC);
+      if (privateList.size() > publicList.size()) {
+        checkNotExistsInBothList(publicList, privateList);
```
Comment (bvinayakumar) [inline common-util/src/main/java/org/sunbird/common/request/UserProfileRequestValidator.java]: Call checkNotExistsInBothList directly from validatePublicAndPrivateFields.

File: common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java [hunk 10]
```diff
@@ -50,21 +48,25 @@ public static void validateCreateUser(Request userRequest) {
...
+  }
   /**
    * This method will validate create user data.
    *
    * @param userRequest Request
    */
-  public static void validateCreateUserV2(Request userRequest) {
+  public void validateCreateUserV2Request(Request userRequest) {
     BaseRequestValidator requestValidator = new BaseRequestValidator();
     requestValidator.validateParam(
```
Comment (bvinayakumar) [inline common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java]: Directly call validateParam

File: common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java [hunk 11]
```diff
@@ -151,6 +151,22 @@ public static void createUserBasicValidation(Request userRequest) {
               ResponseCode.dataTypeError.getErrorMessage(), JsonKey.LANGUAGE, JsonKey.LIST),
           ERROR_CODE);
     }
+  }
+
+  private void createUserBasicProfileFieldsValidation(Request userRequest) {
+    if (StringUtils.isBlank((String) userRequest.getRequest().get(JsonKey.FIRST_NAME))) {
```
Comment (bvinayakumar) [inline common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java]: BaseRequestValidator.validateParam can be used with mandatoryParamsMissing error.

File: common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java [hunk 12]
```diff
@@ -611,6 +624,31 @@ private static void updateUserBasicValidation(Request userRequest) {
             ERROR_CODE);
       }
     }
+    validateLangaugeFields(userRequest);
+  }
+
+  private void validateUserIdOrExternalId(Request userRequest) {
+    if ((StringUtils.isBlank((String) userRequest.getRequest().get(JsonKey.USER_ID))
+            && StringUtils.isBlank((String) userRequest.getRequest().get(JsonKey.ID)))
```
Comment (bvinayakumar) [inline common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java]: Why both USER_ID and ID are mandatory?

File: common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java [hunk 13]
```diff
@@ -709,7 +747,7 @@ public static void validateAssignRole(Request request) {
   }
 
   /** @param request */
-  public static void validateForgotpassword(Request request) {
+  public void validateForgotpassword(Request request) {
```
Comment (bvinayakumar) [inline common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java]: validateForgotPassword

File: common-util/src/test/java/org/sunbird/common/request/UserProfileRequestValidatorTest.java [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+import org.sunbird.common.exception.ProjectCommonException;
+import org.sunbird.common.models.util.JsonKey;
+
+public class UserProfileRequestValidatorTest {
+
+  private static final UserProfileRequestValidator userProfileRequestValidator =
+      new UserProfileRequestValidator();
+
+  @Test
+  public void profileVisibilityValidatorTest() {
```
Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/request/UserProfileRequestValidatorTest.java]: testValidateProfileVisibilityFailureWithFieldInPrivateAndPublic

File: common-util/src/test/java/org/sunbird/common/request/UserProfileRequestValidatorTest.java [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+  }
+
+  @Test
+  public void profileVisibilityInvalidUserTest() {
+    Request request = new Request();
+    Map<String, Object> requestObj = new HashMap<>();
+    requestObj.put(JsonKey.USER_ID, "");
+    request.setRequest(requestObj);
+    try {
+      userProfileRequestValidator.validateProfileVisibility(request);
```
Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/request/UserProfileRequestValidatorTest.java]: testValidateProfileVisibilityFailureWithEmptyUserId

File: common-util/src/test/java/org/sunbird/common/request/UserProfileRequestValidatorTest.java [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+    } catch (ProjectCommonException e) {
+      Assert.assertNotNull(e);
+    }
+  }
+
+  @Test
+  public void profileVisibilityInvalidUserTest() {
```
Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/request/UserProfileRequestValidatorTest.java]: testValidateProfileVisibilityFailureWithEmptyUserId

File: common-util/src/test/java/org/sunbird/common/request/UserProfileRequestValidatorTest.java [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+    request.setRequest(requestObj);
+    try {
+      userProfileRequestValidator.validateProfileVisibility(request);
+    } catch (ProjectCommonException e) {
+      Assert.assertNotNull(e);
+    }
+  }
+
+  @Test
+  public void profileVisibilityInvalidPrivateFieldsTest() {
```
Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/request/UserProfileRequestValidatorTest.java]: testValidateProfileVisibilityFailureWithInvalidPrivateType

File: common-util/src/test/java/org/sunbird/common/request/UserProfileRequestValidatorTest.java [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+    request.setRequest(requestObj);
+    try {
+      userProfileRequestValidator.validateProfileVisibility(request);
+    } catch (ProjectCommonException e) {
+      Assert.assertNotNull(e);
+    }
+  }
+
+  @Test
+  public void profileVisibilityInvalidPublicFieldsTest() {
```
Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/request/UserProfileRequestValidatorTest.java]: testValidateProfileVisibilityFailureWithInvalidPublicType

File: common-util/src/test/java/org/sunbird/common/request/UserRequestValidatorTest.java [hunk 19]
```diff
@@ -1214,4 +1218,37 @@ public void validateGetUserCountTest5() {
       assertEquals(ResponseCode.functionalityMissing.getErrorCode(), e.getCode());
     }
   }
+
+  @Test
+  public void validateVerifyUserTest() {
```
Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/request/UserRequestValidatorTest.java]: testValidateVerifyUserSuccess

File: common-util/src/test/java/org/sunbird/common/request/UserRequestValidatorTest.java [hunk 20]
```diff
@@ -1214,4 +1218,37 @@ public void validateGetUserCountTest5() {
...
+      new UserRequestValidator().validateVerifyUser(request);
+      response = true;
+    } catch (ProjectCommonException e) {
+      Assert.assertNull(e);
+    }
+    Assert.assertTrue(response);
+  }
+
+  @Test
+  public void validateVerifyUserWithOutLoginIdTest() {
```
Comment (bvinayakumar) [inline common-util/src/test/java/org/sunbird/common/request/UserRequestValidatorTest.java]: testValidateVerifyUserFailureWithEmtpyLoginId

File: common-util/src/main/java/org/sunbird/common/request/UserProfileRequestValidator.java (hunk 7 above)
Comment (amolnin) [reply common-util/src/main/java/org/sunbird/common/request/UserProfileRequestValidator.java]: was already there - probably trying to pick a list with lesser size to iterate. 
It won't give much impact so removing.

**Pull request merged**